backend/apis/leads/get-lead.api.js

```diff
@@ -0,0 +1,144 @@
+
+app.get("/api/leads/:id", (req, res, next) => {
+  const validationErrors = checkValid(req.params, validId("id"));
+
+  if (validationErrors.length > 0) {
+    return invalidRequest(res, validationErrors);
+  }
+
+  getLeadById(req.params.id, (err, lead) => {
+    if (err) {
+      return databaseError(req, res, err);
+    }
+
+    if (lead) {
+      res.send({
+        lead
+      });
+    } else {
+      notFound(res, `Could not find lead with id ${req.params.id}`);
+    }
+  });
+});
+
+exports.getLeadById = getLeadById;
+
+function getLeadById(leadId, cbk, connection) {
+  leadId = Number(leadId);
+
+  const getLead = mysql.format(
+    `
+      SELECT
+        leads.id AS leadId,
+        leads.dateOfSignUp,
+        leads.firstContactAttempt,
+        leads.secondContactAttempt,
+        leads.thirdContactAttempt,
+        leads.inactivityReason,
+        leads.firstName,
+        leads.lastName,
+        leads.phone,
+        leads.smsConsent,
+        leads.zip,
+        leads.age,
+        leads.gender,
+        leads.clientId,
+        leads.isDeleted,
+        leads.dateAdded,
+        leads.dateModified,
+        leads.addedBy,
+        leads.modifiedBy,
+        created.firstName AS createdByFirstName,
+        created.lastName AS createdByLastName,
+        modified.firstName AS modifiedByFirstName,
+        modified.lastName AS modifiedByLastName,
+        JSON_ARRAYAGG(
+          JSON_OBJECT(
+            "serviceId", leadServices.serviceId,
+            "serviceName", services.serviceName
+          )
+        ) AS leadServices
+      FROM leads
+        INNER JOIN users created 
+          ON created.id = leads.addedBy
+        INNER JOIN users modified 
+          ON modified.id = leads.modifiedBy
+        INNER JOIN leadServices 
+          ON leadServices.leadId = leads.id
+        INNER JOIN services 
+          ON services.id = leadServices.serviceId
+      WHERE leads.id = ? AND leads.isDeleted = false;
+    `,
+    [leadId]
+  );
+
+  (connection || pool).query(getLead, (err, data, fields) => {
+    if (err) {
+      return cbk(err, data, fields);
+    }
+
+    const bigLeadObj = data[0];
+
+    if (bigLeadObj.length === 0) {
+      return cbk(err, null);
+    }
+
+    const l = bigLeadObj;
+    const leadServices = JSON.parse(l.leadServices);
+
+    const lead = {
+      id: l.leadId,
+      dateOfSignUp: responseDateWithoutTime(l.dateOfSignUp),
+      leadStatus: l.leadStatus,
+      contactStage: {
+        first: l.firstContactAttempt,
+        second: l.secondContactAttempt,
+        third: l.thirdContactAttempt
+      },
+      inactivityReason: l.inactivityReason,
+      firstName: l.firstName,
+      lastName: l.lastName,
+      phone: l.phone,
+      smsConsent: responseBoolean(l.smsConsent),
+      zip: l.zip,
+      age: l.age,
+      gender: l.gender,
+      leadServices: leadServices.map(service => ({
+        id: service.serviceId,
+        serviceName: service.serviceName
```

Comment:
The API docs have the serviceDescription, too. Which is currently not useful at all. Let's maybe remove the serviceDescription from the API docs?